packages/tools/sandbox/src/sandbox.tsx

```diff
@@ -99,8 +121,30 @@ export class Sandbox extends React.Component<ISandboxProps, { isFooterVisible: b
                 this.setState({ isFooterVisible: !this.state.isFooterVisible });
             }
         });
+
+        //
+
+        window.onerror = (error: any) => {
+            this._globalState.onError.notifyObservers({ message: `${error}` });
+            return true;
+        };
+
+        window.onunhandledrejection = (event) => {
+            // eslint-disable-next-line no-console
+            console.error("Unhandled promise rejection:", event.reason);
+
+            return true;
+        };

```

Comment:
in what case will those be triggered?

Reply:
onerror - every error
onunhandledrejection - if an unhandled exception is raised in a Promise. The OBJLoader doesn't  handle this so in case of an exception inside it the app breaks. Thus we need process the unhandled exception with a global handler. A better option would be to handle the exception in the OBJLoader. On contrary there might be code which has the same flaws as the OBJLoader so to be sure nothing breaks the Sandbox app the global onunhandledrejection callback it's not such a bad idea either.